[MMDS][CDAP-13066] Adds pagination to experiments list and models list in experiments

ajainarayanan opened the pull request:
- Adds pagination to 
  - experiments list view
  - models list view
- Adds keyboard shorcut
- Adds updates to url on page change.

JIRA: https://issues.cask.co/browse/CDAP-13066

ajainarayanan committed:
[CDAP-13066] Adds pagination to experiments list and models list in experiments

 - Adds pagination to Experiments list view
 - Adds pagination to Models list in experiment detailed view
 - Adds appropriate actions for pagination + modifies stores to handle pagination in list view and detailed view

tonybach commented on cdap-ui/app/cdap/components/Experiments/DetailedView/index.js:
```diff
@@ -29,13 +32,56 @@ export default class ExperimentDetails extends Component {
+    offset = parseInt(offset, 10);
+    limit = parseInt(limit, 10);
+    if (isNaN(offset)) {
+      offset = DEFAULT_EXPERIMENT_DETAILS.offset;
```
Doesn't seem like there's an `offset` property defined for `DEFAULT_EXPERIMENT_DETAILS`. Did you mean to use `modelsOffset` here?

tonybach commented on cdap-ui/app/cdap/components/Experiments/DetailedView/index.js:
```diff
@@ -29,13 +32,56 @@ export default class ExperimentDetails extends Component {
+      offset = DEFAULT_EXPERIMENT_DETAILS.offset;
+    }
+    if (isNaN(limit)) {
+      limit = DEFAULT_EXPERIMENT_DETAILS.limit;
```
Same here, `modelsLimit`?

tonybach commented:
For some reason, it's showing 'Invalid page' instead of 'Empty page' when I don't have any experiments yet:

![screen shot 2018-01-26 at 11 08 11 am](https://user-images.githubusercontent.com/6516002/35457013-3a946954-028c-11e8-8ede-c471c0988a38.png)

tonybach commented:
So I changed the default models limit per page to 2 for easier testing, then created 3 models. The counts here don't look right:

![model-pagination](https://user-images.githubusercontent.com/6516002/35457034-5003acdc-028c-11e8-9611-8d1d68bcdb7d.gif)

ajainarayanan commented:
@tonybach Have addressed review comments. Issues seeing related to invalid page message is because of a backend issue @CuriousVini is working on. Have fixed the total number of entities next to the pagination.

tonybach commented:
One small problem:

![model-count](https://user-images.githubusercontent.com/6516002/35466505-32a247a2-02b9-11e8-971d-30113b403f29.gif)

Seems like the plurality of the label is still based on the count of models in that page, not on the total count.

tonybach reviewed (approved):
LGTM 👍

ajainarayanan commented:
Reviewed with @bdmogal . Merging.
